.github/workflows/sync_develop_master.yml

```diff
@@ -0,0 +1,75 @@
+name: Branch synchronizer
+
+on:
+  schedule:
+    - cron:  '0 04 * * *'
+
+jobs:
+  sync-master-with-released:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout released
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+          fetch-depth: 0
+          ref: released
+      - name: Check for diff
+        id: check-diff
+        run: |
+          git fetch origin master
+          git checkout master
+          if [ -n "$(git diff master...released)" ]; then
+            echo "Master needs sync with released!";
+            echo "create_pull_request=1" >> $GITHUB_OUTPUT
+          else
+            echo "Master is already up to date.";
+            echo "create_pull_request=0" >> $GITHUB_OUTPUT
+          fi
+      - name: Create Pull Request if necessary
+        if: ${{ steps.check-diff.outputs.create_pull_request == 1 }}
+        uses: repo-sync/pull-request@v2.6.2
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          source_branch: released
+          destination_branch: master
+          pr_title: Merge released into master
+          pr_body: Synchronizes changes of released branch into master branch.
+          pr_reviewer: Maintainers
+          pr_label: "synchronization"
+    outputs:
+      pr_opened: ${{ steps.check-diff.outputs.create_pull_request == 1 }}
+  sync-develop-with-master:
+    runs-on: ubuntu-latest
+    needs: sync-master-with-released
+    steps:
+      - name: Checkout master
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+          fetch-depth: 0
+          ref: master
+      - name: Check for diff
+        id: check-diff
+        run: |
+          git fetch origin develop
+          git checkout develop
+          if [ -n "$(git diff develop...master)" ]; then
+            echo "Develop needs sync with master!";
+            echo "create_pull_request=1" >> $GITHUB_OUTPUT
+          else
+            echo "Develop is already up to date.";
+            echo "create_pull_request=0" >> $GITHUB_OUTPUT
+          fi
+      - name: Create Pull Request if necessary
+        if: ${{ steps.check-diff.outputs.create_pull_request == 1 }} || ${{ needs.sync-master-with-released.outputs.pr_opened == 1 }}
+        uses: repo-sync/pull-request@v2.6.2
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          source_branch: master
+          destination_branch: develop
+          pr_title: Merge master into develop
+          pr_body: Synchronizes changes of master branch into develop branch.
+          pr_reviewer: Maintainers
+          pr_label: "synchronization"
+          pr_allow_empty: true
```

Comment:
I'm not sure about this option. Usually we only push bug fixes to master, meaning "normally" we are much more active on develop rather than master. Does this option entail that every day we'll have an empty PR popping up telling us there's nothing sync?

Reply:
perhaps I misunderstood, the step is run only if there's a diff. In that case then it would never be empty, no?